libs/blocks/columns/columns.js

```diff
@@ -1,19 +1,84 @@
+const getMeaningfulChildren = (col) => Array.from(col.childNodes).filter(
+  (node) => node.nodeType !== Node.TEXT_NODE || node.textContent.trim() !== '',
+);
+
+const isRowHeader = (cols, col, cdx, rows) => {
+  const columnsWithStrongOrH = Array.from(cols)
+    .filter((rowCol) => getMeaningfulChildren(rowCol)
+      .some((child) => child.tagName?.match(/^(H[1-6]|STRONG)$/)));
+
+  const firstRowCols = rows[0].querySelectorAll(':scope > div');
+  const correspondingFirstRowCol = firstRowCols[cdx];
+  const hasColumnHeaderInFirstRow = correspondingFirstRowCol?.getAttribute('role') === 'columnheader';
+
+  return columnsWithStrongOrH.length === 1
+  && columnsWithStrongOrH[0] === col
+  && !hasColumnHeaderInFirstRow;
```

Comment:
Formatting nit, we usually indent multi-line condition checks
```suggestion
  return columnsWithStrongOrH.length === 1
    && columnsWithStrongOrH[0] === col
    && !hasColumnHeaderInFirstRow;
```